shap/explainers/tree.py

```diff
@@ -506,6 +506,17 @@ def __init__(self, model, data=None, data_missing=None):
             self.objective = objective_name_map.get(model.objective, None)
             self.tree_output = tree_output_name_map.get(model.objective, None)
             self.tree_limit = getattr(model, "best_ntree_limit", None)
+        elif str(type(model)).endswith("xgboost.sklearn.XGBRanker'>"):
+            assert_import("xgboost")
+            self.original_model = model.get_booster()
+            self.model_type = "xgboost"
+            xgb_loader = XGBTreeModelLoader(self.original_model)
+            self.trees = xgb_loader.get_trees(data=data, data_missing=data_missing)
+            self.base_offset = xgb_loader.base_score
+            less_than_or_equal = False
+            self.objective = objective_name_map.get(model.objective, None)
```

Comment:
note: for regressor the model.objective is used here, but for classifier xgb_loader.objective is used.  Wondering if it should be made consistent?

Reply:
yeah xgb_loader seems better to me

Reply:
done!